### PR TITLE
[quant][core][improvements] Removed reflection_pad1d_quantized_cpu, dimension and output resizing code in reflection_pad1d_out_template and implemented reflection_pad1d_out_quantized_cpu

### PR DESCRIPTION
**Pull request by dzdang**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #74755

Summary: With the introduction of structured kernel support for quantized tensors in
https://github.com/pytorch/pytorch/pull/74560, we are able to remove the dimension and output
resizing code in reflection_pad1d_out_template. This code is already present in reflection_pad1d.
reflection_pad1d_quantized_cpu has also been removed as quantized tensors can now use reflection_pad1d
after the changes in the linked PR.
reflection_pad1d_out_quantized_cpu was implemented for quantized tensors.

This PR should introduce no functional changes.

Test plan:
```
python run_test.py
```

Differential Revision: [D35148152](https://our.internmc.facebook.com/intern/diff/D35148152)

### CODE REVIEW

**Comment by dzdang** [comment]
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35148152).

**Comment by dzdang** [comment]
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35148152).

**Review by jerryzh168 — APPROVED** [review]
Looks good, thanks!

File: aten/src/ATen/native/ReflectionPad.cpp [hunk 1]
```diff
@@ -264,30 +264,28 @@ inline void reflection_pad1d_out_loop(
...
-  auto pad_r = padding[1];
-
-  int64_t nplane = input_.size(dim_plane);
-  int64_t input_w = input_.size(dim_w);
-  int64_t output_w  = input_w + pad_l + pad_r;
+  pad_l = padding[0];
+  pad_r = padding[1];
+  nplane = input_.size(dim_plane);
+  input_w = input_.size(dim_w);
+  output_w  = input_w + pad_l + pad_r;
```
Comment (jerryzh168) [inline aten/src/ATen/native/ReflectionPad.cpp:288]: actually I don't feel it's a good practice to pass in reference and modify them in the function, can we just copy around this code or have a function that returns these things?

Comment (jerryzh168) [reply aten/src/ATen/native/ReflectionPad.cpp:288]: can structured kernel help here as well? cc @ezyang

Comment (ezyang) [reply aten/src/ATen/native/ReflectionPad.cpp:288]: I have no idea what this code is trying to do, but yes, structured kernels let you precompute quantities like this so they are passed in directly to the function.

**Comment by dzdang** [comment]
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35148152).

**Comment by dzdang** [comment]
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35148152).

**Comment by dzdang** [comment]
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35148152).

**Pull request closed without merging**